lib/render.js

```diff
@@ -25,28 +27,39 @@ function composeData(data) {
 }
 
 /**
- * Take in the data returned from Amphora and turn it into HTML output.
+ * Take in the data returned from Amphora, call `render` hooks,
+ * and return HTML output.
  *
  * @param  {Object} data
- * @return {Promise}
+ * @return {Promise<String>} html
  */
 function render(data) {
-  const template = clayUtils.getComponentName(data._layoutRef || data._self),
-    rootPartial = hbs.partials[template],
-    composedData = composeData(data);
-
-  if (!rootPartial) {
-    throw new Error(`Missing template for ${data.template}`);
-  }
-
-  return Promise.resolve(rootPartial(composedData))
-    .then(mediaService.append(data._media, data.locals.edit, data.locals.site))
-    .then(result => {
-      return {
-        output: result,
-        type: 'html'
-      };
-    });
+
+  const hooks = _.chain(setup.plugins)
+                  .map('render')
```

Comment:
Why `chain --> map --> compact`? That seems just like a `filter` at first glance.

Reply:
`hooks` is an array of `render` functions, so we want to do more than filter

Reply:
You just want to find the plugins that have a `render` function exposed, right? 

`const hooks = _.filter(setup.plugins, plugin => !!plugin.render);